File: Java/day19.java
import java.io.*;
import java.util.*;

interface AdvancedArithmetic {
   int divisorSum(int n);
}

// Write your code here
class Calculator implements AdvancedArithmetic {
    public int divisorSum(int n) {
        int sum = 0;
        if (n >= 1) { sum += n; }
        for (int i = 1; i <= n/2; i++) {
            if (n % i == 0) { sum += i; }
        }
        return sum;
    }
}

public class day19
{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
      	AdvancedArithmetic myCalculator = new Calculator(); 
        int sum = myCalculator.divisorSum(n);
        System.out.println("I implemented: AdvancedArithmetic\n" + sum);
    }
}

File: Python/day19.py
from abc import ABCMeta, abstractmethod

class AdvancedArithmetic():
    __metaclass__ = ABCMeta
    @abstractmethod
    def divisorSum(self, n):
        pass

# Write your code here
class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        sum = 0
        if (n >= 1):
            sum += n
        for i in range(1, (n/2)+1):
            if (n % i == 0):
                sum += i
        return sum

n = int(raw_input())
myCalculator = Calculator()
sum = myCalculator.divisorSum(n)
print "I implemented: AdvancedArithmetic\n", sum